node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -386,8 +386,9 @@ public Node setDirty(String hostname) {
         Node nodeToDirty = getNode(hostname, Node.State.provisioned, Node.State.failed, Node.State.parked).orElseThrow(() ->
                 new IllegalArgumentException("Could not deallocate " + hostname + ": No such node in the provisioned, failed or parked state"));
 
-        if (nodeToDirty.status().hardwareFailureDescription().isPresent())
-            throw new IllegalArgumentException("Could not deallocate " + hostname + ": It has a hardware failure");
+        if (nodeToDirty.status().hardwareFailureDescription().isPresent() || nodeToDirty.status().hardwareDivergence().isPresent())
```

Comment:
This feels wrong to me, I don't know why this was here already, but I would've expected this check to be in `setReady()`. Its strange that an `inactive`/`reserved` node can be denied changing to `dirty` because HW failure/divergence, meanwhile denying changing to `ready` seems reasonable.

Its fine by me since the HW failure check was here already. @hakonhall  / @mpolden  have any opinion on this?

Reply:
This method is only called from the REST API, as a manual operation. In that case it can only be moved from provisioned, failed or parked (the condition above this one) and it doesn't make sense to move a node from those states if it has hardware failure.

Reply:
This check in `setDirty(String hostname)` is only applied to requests from the REST api (i.e. chef), the InactiveExpirer / ReservedExpirer will call `setDirty(Node node)` directly.

Reply:
> as a manual operation

and by Chef, right?

Reply:
Looks like it, but only if the node is one of the allowed states.